### PR TITLE
SALTO-2957 Support filtering by name in all simple adapters

### PR DESCRIPTION
**Pull request by netama**
Support "naive" filtering by the name field (with one exception in workato).
Based on the Jira support.

---

Note that this may not always behave as expected - e.g. filtering out a folder instance does not mean instances that are "contained" in it are removed.

Need to add better tests before merging.

---
_Release Notes_: 

_Zendesk adapter_:
* It is now possible to filter the fetched resources by their name.

_Workato adapter_:
* It is now possible to filter the fetched resources by their name.

_Zuora Billing adapter_:
* It is now possible to filter the fetched resources by their name.

_Stripe adapter_:
* It is now possible to filter the fetched resources by their name.

_Okta adapter_:
* It is now possible to filter the fetched resources by their name.

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/config_doc.md [hunk 1]
```diff
@@ -53,7 +53,6 @@ jira {
 |----------------|------------------------|------------
 | maxAttempts    | `5`                    | The number of attempts to make for each request
 | retryDelay     | `5000` (5 seconds)     | The time (milliseconds) to wait between attempts
-| retryStrategy  | `NetworkError`         | In which cases to retry. Supported choices: `NetworkError` (retry on network errors), `HttpError` (retry on HTTP 5xx errors), or `HTTPOrNetworkError` (both)
```
Comment (alonstern) [inline packages/jira-adapter/config_doc.md:56]: Was it not really supported?
Could you also remove it from the example at the top of the page?

Comment (netama) [reply packages/jira-adapter/config_doc.md:56]: it's not supported (unless I'm missing something), seems it was copied from the salesforce doc at the time. will do

File: packages/okta-adapter/test/fetch_criteria.test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ElemID, InstanceElement, ObjectType } from '@salto-io/adapter-api'
+import fetchCriteria from '../src/fetch_criteria'
+
+describe('fetch_criteria', () => {
+  describe('name', () => {
```
Comment (alonstern) [inline packages/okta-adapter/test/fetch_criteria.test.ts:20]: do we need to test it in every adapter if the implementation is in adapter-components?

File: packages/workato-adapter/src/filters/query.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      element => (
+        isInstanceElement(element)
+        && element.elemID.typeName !== RECIPE_CODE_TYPE
+        && recipeCodesToRemove.has(element.elemID.name)
+      )
+    ))
+
+    if (removedInstances.length > 0) {
+      log.debug(`Omitted ${removedInstances.length} instances that did not match the fetch criteria. The first 100 ids that were removed are: ${removedInstances.slice(0, 100).map(e => e.elemID.getFullName()).join(', ')}`)
+    }
```
Comment (alonstern) [inline packages/workato-adapter/src/filters/query.ts:53]: should we add tests for this part?

**Review by alonstern — APPROVED** [review]
Looks good 👍

File: packages/workato-adapter/src/filters/query.ts (hunk 3 above)
Comment (netama) [reply packages/workato-adapter/src/filters/query.ts:53]: yep, planning to add before merging :)

**Comment by netama** [comment]
replaced by https://github.com/salto-io/salto/pull/4139

**Pull request closed without merging**